index.js

```diff
@@ -74,8 +81,59 @@ router.post('/pointers/:name', async ctx => {
   ctx.body = pointer
 })
 
+// Update revshare pointer
+router.put('/pointers/:name', auth.authorize, async ctx => {
+  const { payout } = ctx.request.body
+
+  if (/[^A-Za-z0-9\-_]/.test(ctx.params.name)) {
+    ctx.throw(400, 'invalid :name. must only use base64url characters.')
+    return
+  }
+
+  try {
+    const pointer = await db.get('pointer:' + ctx.params.name)
+  } catch (e) {
+    debug('failed to lookup pointer. error=' + e.message)
+    ctx.throw(400, 'entry does not exist')
+  }
+
+  let sum = 0
+  for (const entity of payout) {
+    if (typeof entity.percent !== 'number' || isNaN(entity.percent)) {
+      ctx.throw(400, 'invalid percent. must be number.')
+      return
+    }
+
+    if (entity.percent <= 0) {
+      ctx.throw(400, 'invalid percent. must be positive.')
+    }
+
+    sum += entity.percent
+
+    if (!entity.pointer || typeof entity.pointer !== 'string') {
+      ctx.throw(400, 'invalid pointer. must be string.')
+      return
+    }
+  }
+
+  if (sum !== 100) {
+    ctx.throw(400, 'invalid payout. must percents must sum to 100')
+    return
+  }
+
+  const pointer = {
+    name: ctx.params.name,
+    payout: ctx.request.body.payout
+  }
+
+  await db.put('pointer:' + ctx.params.name, JSON.stringify(pointer))
+
+  debug('updated pointer. name=' + ctx.params.name)
+  ctx.body = pointer
+})
+
 // Get details of revshare pointer
-router.get('/pointers/:name', async ctx => {
+router.get('/pointers/:name', auth.authorize, async ctx => {
```

Comment:
Not 100% sure if this endpoint should require auth. Maybe it should be an environment variable whether getting pointer info requires auth? If you make it public then people could look at where their money is going (of course they couldn't trust that info 100% or anything but it'd be cool to see still).

Or maybe we could say that if an environment variable is set then we would set `.receiver_info.payout` in the SPSP query to contain the `payout` object. Then wallets could easily preview that info, which would be very cool

Reply:
I was trying to think whether users would want their payment pointers exposed to the public. If they perceive payment pointers similar to email addresses, maybe not. But on second thought, I think that would depend on the use case, so making auth for this endpoint optional by default and leaving the decision to the discretion of the developer is sensible.So, I'll add a bool type environment variable for whether auth is required and check in the authorize method whether it its set. Only for the get info endpoint.

If set to true, then the process will move forward with authorization. Also, the `.receiver_info.payout` would be set and contain the `payout` object.

If set to false, the process will suspend the authorization middleware and proceed to obtain the pointer info. 

Is this what you have in mind?

Reply:
To keep it simple, I've removed the auth requirement from the get info endpoint. If you feel strongly about adding the environment variable, just let me know. Happy to work on it. Otherwise, I believe that's the last of your outstanding comments. Thanks.